contrib/playlib/src/mill/playlib/Static.scala

```diff
@@ -0,0 +1,103 @@
+package mill.playlib
+
+import java.net.URI
+import java.nio.file._
+import java.nio.file.attribute.BasicFileAttributes
+import java.util
+
+import mill.{PathRef, T}
+import mill.scalalib.{Lib, ScalaModule}
+
+trait Static extends ScalaModule {
+  /**
+    * project resources including configuration, webjars and static assets
+    */
+  override def resources = T.sources {
+    super.resources() :+ webJarResources() :+ staticAssets()
+  }
+
+  /**
+    * Resource base path of packaged assets (path they will appear in in the jar)
+    */
+  def assetsPath = T{ "public" }
+
+  /**
+    *  Directories to include assets from
+    */
+  def assetSources = T.sources{ millSourcePath / assetsPath() }
+
+  /*
+  Collected static assets for the project
+   */
+  def staticAssets = T {
+    val toPath = os.Path(assetsPath(), T.ctx().dest)
+    assetSources().foreach{ pathRef =>
+      val fromPath = pathRef.path
+      if (os.isDir(fromPath)) {
+        os.walk(fromPath).filter(os.isFile(_)).foreach{ p =>
+          os.copy(p, toPath / p.relativeTo(fromPath), createFolders = true)
+        }
+      }
+    }
+    PathRef(T.ctx().dest)
+  }
+
+  /**
+    * webjar dependencies - created from transitive ivy deps
+    */
+  def webJarDeps = T{
+    transitiveIvyDeps().filter(_.dep.module.organization == "org.webjars")
+  }
+
+  /**
+    * jar files of web jars
+    */
+  def webJars = T{
+    Lib.resolveDependencies(repositories, Lib.depToDependency(_, scalaVersion()), webJarDeps())
+  }
+
+  /**
+    * webjar resources extracted from their source jars with version from path removed
+    */
+  def webJarResources = T {
+    extractWebJars(webJars().toSeq, os.Path(assetsPath(), T.ctx().dest) / 'lib)
+    PathRef(T.ctx().dest)
+  }
+
+
+  def start(args: String*) = T.command{ run(args:_*) }
```

Comment:
I don't think this was meant to be extracted

Reply:
it wasn't this is now fixed, thanks